simd/unit_tests/TestSIMD.cpp

```diff
@@ -277,6 +277,30 @@ class divides {
   }
 };
 
+class shift_right {
+ public:
+  template <typename T>
+  auto on_host(T&& a, unsigned int b) const {
+    return a >> b;
+  }
+  template <typename T>
+  KOKKOS_INLINE_FUNCTION auto on_device(T& a, unsigned int b) const {
+    return a >> b;
+  }
+};
```

Comment:
Where do we test when the rhs is a simd type?

Reply:
Added test cases that exercise shift ops with simd type rhs in [2aa01ef](https://github.com/kokkos/kokkos/pull/6109/commits/2aa01ef5b992084bda24cb977cb3f485b26de0ec).
Device side ones added in [775b605](https://github.com/kokkos/kokkos/pull/6109/commits/775b6056351fb7edaee60fdae6ff951b94591378).